ddprof-lib/src/main/cpp/profiler.cpp

```diff
@@ -996,21 +997,21 @@ void Profiler::updateJavaThreadNames() {
 }
 
 void Profiler::updateNativeThreadNames() {
-  ThreadList *thread_list = OS::listThreads();
-  for (int tid; (tid = thread_list->next()) != -1;) {
-    _thread_info.updateThreadName(
-        tid, [](int tid) -> std::unique_ptr<char[]> {
-          const size_t buffer_size = 64;
-          char *name_buf = new char[buffer_size];
-          if (OS::threadName(tid, name_buf, buffer_size)) {
-            return std::unique_ptr<char[]>(name_buf);
-          }
-          delete[] name_buf;
-          return nullptr;
-        });
-  }
+    ThreadList *thread_list = OS::listThreads();
+    constexpr size_t buffer_size = 64;
+    char name_buf[buffer_size];  // Stack-allocated buffer
+
+    for (int tid; (tid = thread_list->next()) != -1;) {
+        _thread_info.updateThreadName(
+                tid, [&](int tid) -> std::string {
+                    if (OS::threadName(tid, name_buf, buffer_size)) {
+                        return std::string(name_buf, buffer_size);
```

Comment:
Yes, for anything that is large, there will be an allocation, however, it will be moved into the map on the other side (so in all, there should be one less allocation).

Reply:
To be clear, before we were allocating in a loop, then we were copying the C buffer into the std::string (creating an extra copy). Now we have a single string creation.

Reply:
Yes, makes sense.